tests/test_multistore_file.py

```diff
@@ -0,0 +1,40 @@
+"""Unit tests for oauth2client.multistore_file."""
+
+import errno
+import os
+import tempfile
+import unittest
+
+from oauth2client import multistore_file
+
+
+class _MockLockedFile(object):
+
+    def __init__(self, filename_str, error_code):
+        self.filename_str = filename_str
+        self.error_code = error_code
+        self.open_and_lock_called = False
+
+    def open_and_lock(self):
+        self.open_and_lock_called = True
+        raise IOError(self.error_code, '')
+
+    def is_locked(self):
+        return False
+
+    def filename(self):
+        return self.filename_str
+
+
+class MultistoreFileTests(unittest.TestCase):
+
+    def test_lock_file_raises_ioerror(self):
+        filehandle, filename = tempfile.mkstemp()
+        os.close(filehandle)
```

Comment:
It depends on where those authors are coming from; either way, both functions are documented in the [os module documentation](https://docs.python.org/2/library/os.html#os.unlink) and no preference is stated there.

The I/O is not unnecessary - the documentation for mkstemp explicitly states that it is the user's responsibility to clean it.  While it may currently be in a temporary directory that's cleaned by the OS, that's not guaranteed in the future. 

These unit tests already involve many file accesses, so I do not think that this will impact performance in a meaningful way.  If you disagree, provide some data explaining the unit test performance targets?